2020/09/index.js

```diff
@@ -1,3 +1,97 @@
+const { promises: fs } = require('fs');
+
+async function getInput() {
+  return input = (await fs.readFile(`${__dirname}/input.txt`))
+    .toString()
+    .trim()
+    .split('\n')
+    .map(i => parseInt(i));
+}
+
+class OrderedSet {
```

Comment:
A misnomer perhaps. I wanted something with `O(1)` lookup while tracking the order they were inserted so I can remove the earliest entry.